app/models/compute_resources/foreman/model/libvirt.rb

```diff
@@ -289,7 +292,9 @@ def vm_instance_defaults
         :display    => { :type     => display_type,
                          :listen   => Setting[:libvirt_default_console_address],
                          :password => random_password(console_password_length(display_type)),
-                         :port     => '-1' }
+                         :port     => '-1' },
+        :firmware   => 'automatic',
+        :firmware_features => { "secure-boot" => "no" }
```

Comment:
Is there a reason for having strings as keys in the inner hash?

Reply:
Yes, the reason for using strings as keys here is that Libvirt expects these values in this format and converts them to XML accordingly. You can see this conversion in the Libvirt code [here](https://github.com/fog/fog-libvirt/blob/master/lib/fog/libvirt/models/compute/server.rb#L277). For more details, refer to the related [PR](https://github.com/fog/fog-libvirt/pull/155).